src/write.c

```diff
@@ -142,8 +158,16 @@ static avifEncoderItem * avifEncoderDataCreateItem(avifEncoderData * data, const
     item->infeNameSize = infeNameSize;
     item->encodeOutput = avifCodecEncodeOutputCreate();
     item->cellIndex = cellIndex;
-    avifArrayCreate(&item->mdatFixups, sizeof(avifOffsetFixup), 4);
+    if (!avifArrayCreate(&item->mdatFixups, sizeof(avifOffsetFixup), 4)) {
```

Comment:
optional: if you move this to line 154 (immediately after *item is populated for the first time), the error handling code can be simplified to simply pop the array element instead of having to revert a bunch of other things.

Reply:
Thank you for the suggestion. We still need to handle the failure of `item->encodeOutput = avifCodecEncodeOutputCreate()` at line 159, so in this pull request I will just move the `++data->lastItemID` operation right before the successful return.